roles/cnf_cert/tasks/pre-run.yml

```diff
@@ -155,12 +155,15 @@
         version: "{{ test_network_function_version }}"
         dest: "{{ tnf_dir }}"
         force: true
+      environment:
+        GIT_HTTP_MAX_REQUESTS: "4"
+        GIT_HTTP_IDLE_TIMEOUT: "3600"
```

Comment:
my Google fu is too basic that I couldn't find these variables 😛, I was just wondering what it means. Is this 4 retries of potentially 1 hour timeout each ? Or 4 request in a full hour, if the later then 15 mins looks good to me.

Reply:
@tkrishtop I think we are only interested in the latest commit and do not care about the whole history. This will significantly reduce the data that needs to be transferred. 

Do you think using shallow here will work?

 depth: 1

Create a shallow clone with a history truncated to the specified number or revisions. The minimum possible value is 1, otherwise ignored. Needs git>=1.9.1 to work correctly.

```
$ git clone --depth=1 --origin origin https://github.com/test-network-function/cnf-certification-test
Cloning into 'cnf-certification-test'...
remote: Enumerating objects: 448, done.
remote: Counting objects: 100% (448/448), done.
remote: Compressing objects: 100% (398/398), done.
remote: Total 448 (delta 85), reused 190 (delta 24), pack-reused 0
Receiving objects: 100% (448/448), 716.87 KiB | 1.59 MiB/s, done.
Resolving deltas: 100% (85/85), done.
$ du -hs cnf-certification-test/
6.2M	cnf-certification-test/
```

Reply:
It does not seem to be a problem with the size as it is, we really have some bytes missing
```
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 769 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
```
It's more poor internet connection.

I found one more idea in [my google](https://docs.gitlab.com/ee/topics/git/troubleshooting_git.html#rpc-failed-curl-92-http2-stream-0-was-not-closed-cleanly-internal_error-err-2) though:
```
RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: INTERNAL_ERROR (err 2)
This problem may be caused by a slow internet connection. 
If you use Git over HTTP instead of SSH, try one of these fixes:

Increase the POST buffer size in the Git configuration with git config http.postBuffer 52428800.
Switch to the HTTP/1.1 protocol with git config http.version HTTP/1.1.
If neither approach fixes the error, you may need a different internet service provider.
```

Reply:
@betoredhat and @manurodriguez thank you for the reviews! I decided not to complicate life and went with the solution of Beto, adding `depth=1`  and `reference: "{{ tnf_dir }}"` as the clone options. The main issue is a slow network while cloning the large repo, so let's see, if that may fix the issue.